torchtext/experimental/vocab.py

```diff
@@ -19,7 +19,7 @@
 logger = logging.getLogger(__name__)
 
 
-def build_vocab_from_text_file(file_path, jited_tokenizer, min_freq=1, unk_token='<unk>', num_cpus=4):
```

Comment:
Sounds good! I think this need to be crafted carefully together with default index. Because User (based on their prior experience) may expect that we return index of <unk> token if the query is OOV, which is no longer true (we raise error if default index is not set and OOV is queried). In this new design <unk> is just like any other token for Vocab class.

As as aside, if we say that <unk> is just any other valid token, I guess we can also think of facilitating addition of special  tokens in factory functions?